app/models/transaction.rb

```diff
@@ -26,7 +28,12 @@ class Transaction < ActiveRecord::Base
   delegate :income?, :expense?, to: :category, allow_nil: true
 
   default_scope { order(created_at: :desc) }
-  scope :by_currency, ->(currency) { joins(:bank_account).where('bank_accounts.currency' => currency) }
+  scope :by_currency, ->(currency) { joins("INNER JOIN bank_accounts bank_account_transactions
+      ON bank_account_transactions.id = transactions.bank_account_id
+      AND transactions.deleted_at IS NULL
+      AND bank_account_transactions.deleted_at IS NULL
+      AND bank_accounts.currency = '#{currency}'") }
+  scope :without_hidden, -> { where('bank_accounts.hidden' => false) }
```

Comment:
@Gataniel you don't need this but in any case this scope should be written as:
```rb
scope :without_hidden, -> { joins(:bank_account).where(bank_accounts: { hidden: false })
```
or
```rb
scope :without_hidden, -> { joins(:bank_account).merge(BankAccount.without_hidden)
```